### ISSUE TITLE
Export ProjectSchema

### ISSUE THREAD

**Issue by ruckc** [jdalrymple/gitbeaker#1813]
**Description**

Using 29.2.0 with typescript, when trying to pass the result of `api.Projects.all()` typescript complains about the type of ProjectSchema[] which I can't import (because it isn't exported fully).

**Possible fixes**

I'd like ProjectSchema and any other interfaces to be exported for type resolution.

**Comment by soullivaneuh** [comment]
I think the issue is bigger than that. Since `v29.2.4` update, we don't have any typing of the API structure:

```typescript
  // gitlab.Releases does not match any typing definition.
  gitlab.Releases
    .all(projectId, {
      perPage: 1,
      maxPages: 1,
    })
    // error TS7006: Parameter 'response' implicitly has an 'any' type.
    .then((response) => response[0].tag_name as string)
```

**Comment by jdalrymple** [comment]
Yea there was an issue with the switch to esbuild/dts, reverting those changes with #1851

**Comment by jdalrymple** [comment]
Should be fixed by #1818
